jina/executors/encoders/image/customtfkeras.py

```diff
@@ -0,0 +1,43 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import numpy as np
+
+from ..frameworks import BaseCVTFEncoder
+from ...decorators import batching, as_ndarray
+
+
+
+class CustomImageKerasEncoder(BaseCVTFEncoder):
+    """
+    :class:`CustomImageKerasEncoder` encodes data from a ndarray, potentially B x (Channel x Height x Width) into a
+        ndarray of `B x D`.
+    Internally, :class:`CustomImageKerasEncoder` wraps any custom tf.keras model not part of models from `tensorflow.keras.applications`.
+    https://www.tensorflow.org/api_docs/python/tf/keras/applications
+    """
+
+    def __init__(self, model_path: str, layer_name: str, *args, **kwargs):
+        """
+        :param model_path: the path where the model is stored.
+        :layer: Name of the layer from where to extract the feature map.
+        """
+        super().__init__(*args, **kwargs)
+        self.model_path = model_path
+        self.layer_name = layer_name
+
+    def post_init(self):
+        import tensorflow as tf
```

Comment:
It's better to support both CPUs and GPUs.
```suggestion
        self.to_device()
        import tensorflow as tf
```